src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1435,13 +1435,23 @@ private static LambdaConversionResult IsAnonymousFunctionCompatibleWithDelegate(
 
                 if (anonymousFunction.HasExplicitlyTypedParameterList)
                 {
+                    var lambdaSymbol = anonymousFunction.TemporaryLambdaSymbol;
+
+                    Debug.Assert(lambdaSymbol is not null);
+
                     for (int p = 0; p < delegateParameters.Length; ++p)
                     {
                         if (delegateParameters[p].RefKind != anonymousFunction.RefKind(p) ||
                             !delegateParameters[p].Type.Equals(anonymousFunction.ParameterType(p), TypeCompareKind.AllIgnoreOptions))
                         {
                             return LambdaConversionResult.MismatchedParameterType;
                         }
+
+                        if (lambdaSymbol.Parameters[p].HasExplicitDefaultValue &&
+                            delegateParameters[p].ExplicitDefaultConstantValue != lambdaSymbol.Parameters[p].ExplicitDefaultConstantValue)
+                        {
+                            return LambdaConversionResult.MismatchedParameterDefaultValue;
```

Comment:
>MismatchedParameterDefaultValue [](http://example.com/codeflow?start=58&length=31)

It looks like this change will affect overload resolution. Consider adding a test if there's not one already. For instance:
```csharp
delegate void D1(int i = 1);
delegate void D2(int i = 2);
class Program
{
    static int F(D1 d) => 1;
    static object F(D2 d) => 2;
    static void Main()
    {
        int y = F((int x = 2) => { });
    }
}
```